src/utils/focus/focusable.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * CSS selector to query focusable elements.
+ *
+ * This does not eliminate the following elements which are not focusable:
+ *  - Custom elements with `tabindex` or `contenteditable` attribute
+ *  - Shadow hosts with `delegatesFocus: true`
+ */
+export const FOCUSABLE_SELECTOR = [
+  'input:not([type=hidden]):not([disabled])',
+  'button:not([disabled])',
+  'select:not([disabled])',
+  'textarea:not([disabled])',
+  '[contenteditable=""]',
+  '[contenteditable="true"]',
+  'a[href]',
+  '[tabindex]:not([disabled])',
+].join(', ')
+
+/**
+ * Determine if an element can be the target for `focusElement()`.
+ *
+ * This does not necessarily mean that this element will be the `activeElement`,
+ * as it might delegate focus into a shadow tree.
+ */
+export function isFocusTarget(element: Element): element is HTMLElement {
+  if (element.tagName.includes('-')) {
```

Comment:
I dunno but I think having this a separate helper function might be helpful. And makes it more clear as well.